icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6741,3 +6742,11 @@ static void TestBug20149() {
         log_err("ERROR: in uloc_forLanguageTag %s return %s\n", myErrorName(status), locale);
     }
 }
+
+// For the C locale, ICU should have a default locale of en_US_POSIX
+static void TestCLocale(){
+    const char *defaultLocale = uloc_getDefault();
+    if ( (strcmp(getenv("LANG"), "C") == 0 || strcmp(getenv("LANG"), "C.UTF-8") == 0) && strcmp(defaultLocale, "en_US_POSIX") != 0) {
```

Comment:
`getenv("LANG")` can return `NULL` if the variable isn't set.
We should only call `getenv("LANG")` once, and assign it to variable, then check if that is NULL first.
Reusing the variable also means we don't call `getenv("LANG")` up to 3 times, which is wasteful.